include/deal.II/base/tensor.h

```diff
@@ -1369,8 +1408,8 @@ Tensor<rank_, dim, Number>::unrolled_to_component_indices(const unsigned int i)
   unsigned int remainder = i;
   for (int r = rank_ - 1; r >= 0; --r)
     {
-      indices[r] = (remainder % dim);
-      remainder /= dim;
+      indices[r] = internal::mod<dim>(remainder);
+      internal::div<dim>(remainder);
```

Comment:
and then do
```
  remainder = internal::div<dim>(remainder);
```
here. I think that would be easier to read. It would also restore the symmetry between the `mod` and `div` operations in that both of them return things by value.